.github/workflows/coverage.yml

```diff
@@ -18,9 +18,12 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - uses: actions-rs/toolchain@v1
+      - name: Read rust-toolchain file
+        id: rust-toolchain
+        run: echo "toolchain=$(cat rust-toolchain)" >> $GITHUB_OUTPUT
```

Comment:
Currently, the workflow reads the `rust-toolchain` file directly for usage. This restricts its scalability for changes like using the `rust-toolchain.toml` file instead. I'll find a better solution and let you know (maybe create a new action).